### PR TITLE
document that services healthcheckNodePort is inmutable once set

### LINKED ISSUE #111518: Changing healthCheckNodePort after creating LoadBalancer Service with externalTraffic=Local
### What would you like to be added?

We are currently using v1.19 and it seems that if you have a `LoadBalancer` service with `externalTrafficPolicy` set to `Local` then changing the health check node port is not supported. One way to get around this is by changing the `externalTrafficPolicy` to `Cluster` and then changing it back to `Local` in addition to specifying the `healthCheckNodePort` with the desired port number. 

This behavior is not very well documented and the below error message is not very useful:
```
cannot change healthCheckNodePort on loadBalancer service with externalTraffic=Local during update
```

I wonder if we could also add another case below to release the current health check port and allow changing it to the user specified value given it's valid?

https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/registry/core/service/storage/rest.go#L324-L353

### Why is this needed?

It's possible that after creating a `LoadBalancer` someone wants to change it to a user-provided `healthCheckNodePort` and the current way to do it is only by toggling the `externalTrafficPolicy`. This feature with the documentation would be more straightforward way to make any port changes.

### PR DESCRIPTION
**Pull request by aojea**
/kind documentation
Fixes #111518

```release-note
NONE
```

/assign @thockin

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -4484,6 +4484,7 @@ type ServiceSpec struct {
 	// service or not.  If this field is specified when creating a Service
 	// which does not need it, creation will fail. This field will be wiped
 	// when updating a Service to no longer need it (e.g. changing type).
+	// This field can not be updated once set.
```
Comment (thockin) [inline staging/src/k8s.io/api/core/v1/types.go:4487]: nit: "cannot" is less ambiguous than "can not".  English :shrug:.

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by thockin** [comment]
/lgtm

not sure why I removed that - mistake

**Comment by aojea** [comment]
/hold

the verify errror seems legit, this can wait for 1.26

**Comment by aojea** [comment]
> /hold
> 
> the verify errror seems legit, this can wait for 1.26

/hold cancel
update openapi

**Pull request merged**